### PR TITLE
fix(ManageNetwork): replace Range with RangeInclusive

### PR DESCRIPTION
**Pull request by niklasad1**
Fixes `TODO: Range should be changed to RangeInclusive once stable (https://github.com/rust-lang/rust/pull/50758)`

### CODE REVIEW

File: ethcore/sync/src/api.rs [hunk 1]
```diff
@@ -615,9 +615,7 @@ pub trait ManageNetwork : Send + Sync {
 	/// Stop network
 	fn stop_network(&self);
 	/// Returns the minimum and maximum peers.
-	/// Note that `range.end` is *exclusive*.
-	// TODO: Range should be changed to RangeInclusive once stable (https://github.com/rust-lang/rust/pull/50758)
-	fn num_peers_range(&self) -> Range<u32>;
+	fn num_peers_range(&self) -> RangeInclusive<u32>;
 	/// Get network context for protocol.
```
Comment (niklasad1) [inline ethcore/sync/src/api.rs]: IMHO, return type `RangeInclusive` makes it clear that the range is inclusive thus I removed the comment

File: ethcore/sync/src/api.rs [hunk 2]
```diff
@@ -948,12 +946,12 @@ impl LightSyncProvider for LightSync {
...
 		let (connected, active) = self.proto.peer_count();
 		let peers_range = self.num_peers_range();
-		debug_assert!(peers_range.end > peers_range.start);
+		debug_assert!(peers_range.end() > peers_range.start());
 		PeerNumbers {
 			connected: connected,
 			active: active,
-			max: peers_range.end as usize - 1,
-			min: peers_range.start as usize,
+			max: *peers_range.end() as usize - 1,
```
Comment (niklasad1) [inline ethcore/sync/src/api.rs]: Not addressed by this PR: why do we subtract `end` by 1?

File: ethcore/sync/src/api.rs [hunk 3]
```diff
@@ -948,12 +946,12 @@ impl LightSyncProvider for LightSync {
 	fn peer_numbers(&self) -> PeerNumbers {
 		let (connected, active) = self.proto.peer_count();
 		let peers_range = self.num_peers_range();
-		debug_assert!(peers_range.end > peers_range.start);
+		debug_assert!(peers_range.end() > peers_range.start());
```
Comment (ordian) [inline ethcore/sync/src/api.rs]: Should it be `>=` here and below?

Comment (niklasad1) [reply ethcore/sync/src/api.rs]: Yeah, I think so otherwise this fail if `start == end` because end was exclusive before

**Pull request merged**